src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/bolt/IslCacheBolt.java

```diff
@@ -124,28 +125,26 @@ protected void handleInput(Tuple input) throws PipelineException {
             String flowId = pullValue(input, FLOW_ID_FIELD, String.class);
             FlowDirection direction = pullValue(input, FLOW_DIRECTION_FIELD, FlowDirection.class);
             List<Link> flowPath = (List<Link>) pullValue(input, FLOW_PATH_FIELD, List.class);
-            Long maxLatency = pullValue(input, MAX_LATENCY_FIELD, Long.class);
-            Long maxLatencyTier2 = pullValue(input, MAX_LATENCY_TIER_2_FIELD, Long.class);
 
-            long latency = islCacheService.calculateLatencyForPath(flowPath);
+            Duration latency = Duration.ofNanos(islCacheService.calculateLatencyForPath(flowPath));
 
-            emit(ACTION_STREAM_ID.name(), input, new Values(flowId, direction, latency, maxLatency, maxLatencyTier2,
-                    getCommandContext()));
+            emit(ACTION_STREAM_ID.name(), input, new Values(flowId, direction, latency, getCommandContext()));
             emitLatencyStats(input, flowId, direction, latency);
         } else {
             unhandledInput(input);
         }
     }
 
-    private void emitLatencyStats(Tuple input, String flowId, FlowDirection direction, long latency) {
+    private void emitLatencyStats(Tuple input, String flowId, FlowDirection direction, Duration latency) {
         Map<String, String> tags = ImmutableMap.of(
                 "flowid", flowId,
                 "direction", direction.name().toLowerCase(),
                 "calculated", "true"
         );
 
+        long latencyNano = latency.getSeconds() * TEN_TO_NINE + latency.getNano();
```

Comment:
It can be simplified https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toNanos--

Reply:
Done.